src/root/views/FinalReportForm/useDreFinalReportOptions.ts

```diff
@@ -0,0 +1,343 @@
+  positiveIntegerCondition,
+} from '#utils/form';
+
+import {
+  BooleanValueOption,
+  DrefFinalReportFields,
+  emptyNumericOptionList,
+  emptyStringOptionList,
+  NumericKeyValuePair,
+  NumericValueOption,
+  StringKeyValuePair,
+  User,
+} from './common';
+
+export type FormSchema = ObjectSchema<PartialForm<DrefFinalReportFields>>;
+export type FormSchemaFields = ReturnType<FormSchema['fields']>;
+
+export type CountryDistrictType = NonNullable<NonNullable<DrefFinalReportFields['country_district']>>[number];
+export type CountryDistrictSchema = ObjectSchema<PartialForm<CountryDistrictType>>;
+export type CountryDistrictSchemaFields = ReturnType<CountryDistrictSchema['fields']>;
+export type CountryDistrictsSchema = ArraySchema<PartialForm<CountryDistrictType>>;
+export type CountryDistrictsSchemaMember = ReturnType<CountryDistrictsSchema['member']>;
+
+export type NeedType = NonNullable<NonNullable<DrefFinalReportFields['needs_identified']>>[number];
+export type NeedSchema = ObjectSchema<PartialForm<NeedType>>;
+export type NeedSchemaFields = ReturnType<NeedSchema['fields']>;
+export type NeedsSchema = ArraySchema<PartialForm<NeedType>>;
+export type NeedsSchemaMember = ReturnType<NeedsSchema['member']>;
+
+export type InterventionType = NonNullable<NonNullable<DrefFinalReportFields['planned_interventions']>>[number];
+export type InterventionSchema = ObjectSchema<PartialForm<InterventionType>>;
+export type InterventionSchemaFields = ReturnType<InterventionSchema['fields']>;
+export type InterventionsSchema = ArraySchema<PartialForm<InterventionType>>;
+export type InterventionsSchemaMember = ReturnType<InterventionsSchema['member']>;
+
+export type IndicatorType = InterventionType['indicators'][number];
+export type IndicatorSchema = ObjectSchema<PartialForm<IndicatorType>>;
+export type IndicatorSchemaFields = ReturnType<IndicatorSchema['fields']>;
+export type IndicatorsSchema = ArraySchema<PartialForm<IndicatorType>>;
+export type IndicatorsSchemaMember = ReturnType<IndicatorsSchema['member']>;
+
+
+export const MaxIntLimit = 2147483647;
+
+export const schema: FormSchema = {
+  fields: (value): FormSchemaFields => ({
+    title: [],
+    national_society: [],
+    disaster_type: [],
+    disaster_category: [],
+    type_of_onset: [],
+    country_district: {
+      keySelector: (c) => c.clientId as string,
+      member: (): CountryDistrictsSchemaMember => ({
+        fields: (): CountryDistrictSchemaFields => ({
+          clientId: [],
+          country: [requiredCondition, (value, allValues) => {
+            if (isNotDefined(value)) {
+              return undefined;
+            }
+            const countriesWithCurrentId = (allValues as unknown as DrefFinalReportFields)?.country_district?.filter(
+              d => d.country === value
+            );
+
+            if (countriesWithCurrentId.length > 1) {
+              return 'Duplicate countries not allowed';
+            }
+
+            return undefined;
+          }],
```

Comment:
Let's move this to validation. This will be called for all of the countries. @frozenhelium